### DISCUSSION TITLE
Suggestions on scanning third-party crates.

### DISCUSSION THREAD

**Discussion by xuxiaocheng0201** [fzyzcjy/flutter_rust_bridge#2356] [Ideas]
I have always felt that third-party crates are uncontrollable. A best practice should be to only scan the code in the `rust_root`(default is `rust`) directory.
Another suggestion is using [`o2o`](https://crates.io/crates/o2o) crate, this is a 'Object to Object mapper'. By using this, we can simply map a third-party struct to the suitable struct written by hand. This avoids scanning third-party crates and enhances reliability.

**Comment by fzyzcjy** [comment]
IIRC yes, unless you explicitly config to scan a 3rd party crate, then only 1st party crate is scanned. https://cjycode.com/flutter_rust_bridge/guides/third-party
